frontend/src/features/research/list/ResearchListView.test.tsx

```diff
@@ -0,0 +1,553 @@
+                isDisabled: false,
+              },
+              dataSourceEffectiveDate: '2021-08-31T00:00:00',
+              isSensitive: false,
+              pid: '011-041-404',
+              pin: 90072652,
+              landArea: 1,
+              rowVersion: 2,
+            } as any,
+            rowVersion: 0,
+          },
+        ],
+      },
+    ]);
+    const { container, searchButton, findAllByText } = setup();
+
+    fillInput(container, 'regionCode', 'South Coast Region', 'select');
+    await act(async () => userEvent.click(searchButton));
+
+    expect(await findAllByText(/Southern Interior Region/i)).toHaveLength(2);
+  });
+
+  it('all  unique regions are listed', async () => {
+    setupMockSearch([
+      {
+        id: 1,
+        researchFileStatusTypeCode: {
+          id: 'ACTIVE',
+          description: 'Active',
+          isDisabled: false,
+        },
+        name: 'name',
+        rfileNumber: 'R100-100-100',
+        appLastUpdateUserid: '',
+        appCreateUserid: '',
+        appCreateTimestamp: '2020-01-01',
+        appLastUpdateTimestamp: '2021-01-01',
+        researchProperties: [
+          {
+            id: 1,
+            isDisabled: false,
+            property: {
+              id: 1,
+              region: {
+                id: 1,
+                description: 'Southern Interior Region',
+                isDisabled: false,
+              },
+              dataSourceEffectiveDate: '2021-08-31T00:00:00',
+              isSensitive: false,
+              pid: '007-723-385',
+              pin: 90069930,
+              landArea: 1,
+              rowVersion: 2,
+            },
+            rowVersion: 0,
+          } as any,
+          {
+            id: 2,
+            isDisabled: false,
+            property: {
+              id: 2,
+              region: {
+                id: 2,
+                description: 'South Coast Region',
+                isDisabled: false,
+              },
+              dataSourceEffectiveDate: '2021-08-31T00:00:00',
+              isSensitive: false,
+              pid: '011-041-404',
+              pin: 90072652,
+              landArea: 1,
+              rowVersion: 2,
+            } as any,
+            rowVersion: 0,
+          },
+        ],
+      },
+    ]);
+    const { container, searchButton, findByText } = setup();
+
+    fillInput(container, 'regionCode', 'South Coast Region', 'select');
+    await act(async () => userEvent.click(searchButton));
+
+    expect(await findByText(/Southern Interior Region, South Coast Region/i)).toBeInTheDocument();
+  });
+
+  it('searches by R-file number', async () => {
+    setupMockSearch([]);
+    const { container, searchButton } = setup();
+    fillInput(container, 'researchSearchBy', 'rFileNumber', 'select');
+    fillInput(container, 'rFileNumber', '101');
+    await act(async () => userEvent.click(searchButton));
+
+    expect(getResearchFiles).toHaveBeenCalledWith(expect.objectContaining({}));
```

Comment:
I think this test will always pass? shouldn't we include actual fields being submitted here?

Reply:
yes, went back and forth on these a few times, will correct.